lib/superset/dashboard/export.rb

```diff
@@ -45,35 +65,74 @@ def params
+        puts "Unzipped files: #{@extracted_files.inspect}"
       end
 
       def download_folder
         File.dirname(extracted_files[0])
       end
 
+      def destination_path_with_dash_id
+        @destination_path_with_dash_id ||= File.join(destination_path, dashboard_id.to_s)
+      end
+
       def copy_export_files_to_destination_path
-        path_with_dash_id = File.join(destination_path, dashboard_id.to_s)
-        FileUtils.mkdir_p(path_with_dash_id) unless File.directory?(path_with_dash_id)
+        FileUtils.mkdir_p(destination_path_with_dash_id) unless File.directory?(destination_path_with_dash_id)
+        puts "Ensured destination directory: #{destination_path_with_dash_id}"
+
+        Dir.glob("#{download_folder}/**/*").each do |item|
+          next if File.directory?(item) # Skip directories
+
+          relative_item_path = Pathname.new(item).relative_path_from(Pathname.new(download_folder)).to_s
+          destination_item = File.join(destination_path_with_dash_id, relative_item_path)
 
-        Dir.glob("#{download_folder}/*").each do |item|
-          FileUtils.cp_r(item, path_with_dash_id)
+          FileUtils.mkdir_p(File.dirname(destination_item)) unless File.directory?(File.dirname(destination_item))
+          puts "Copying #{item} to #{destination_item}"
+          FileUtils.cp(item, destination_item) # Using cp instead of cp_r
         end
       end
 
-      def zip_file_name
-        @zip_file_name ||= "#{tmp_uniq_dashboard_path}/dashboard_#{dashboard_id}_export_#{datestamp}.zip"
+      def cleanup_outdated_files
+        destination = Pathname.new(destination_path_with_dash_id)
+        puts "Cleaning up outdated files in #{destination}"
+
+        # Gather a list of relative paths from the download_folder, including only files
+        latest_files = Dir.glob("#{download_folder}/**/*").select { |f| File.file?(f) }.map do |file|
+          Pathname.new(file).relative_path_from(Pathname.new(download_folder)).to_s.downcase
+        end
+        puts "Latest files: #{latest_files.inspect}"
+
+        # Convert latest_files to a Set for efficient lookup
+        latest_files_set = latest_files.to_set
+
+        # Iterate over existing files in the destination path and delete any that are not in the latest export
+        Dir.glob("#{destination}/**/*").each do |existing_file|
+          existing_path = Pathname.new(existing_file)
+          next if existing_path.directory? # Skip directories
+
+          relative_path = existing_path.relative_path_from(destination).to_s.downcase
+          puts "Checking file: #{relative_path}"
+
+          unless latest_files_set.include?(relative_path)
+            puts "Removing file: #{existing_file} as it's not in the latest export."
+            FileUtils.rm_f(existing_file)
+          else
+            puts "Retaining file: #{existing_file} as it's part of the latest export."
+          end
+        end
```

Comment:
@barrynguyen8  there is a fair chunk of logic here. 
I had an idea .....

Q: Given we always copy over the entire new extracted file ... 
do we care about any thing that is in the destination path prior to the copy?

ie ... we always copy over the current state of a dashboard, where all files are considered necessary.
So why not just delete the entire contents of the destination path before we copy across?
No checks / comparisons needed.

```
# something like
Dir.glob("#{destination_path_with_dash_id}/*").each do |file|
  FileUtils.rm(file) if File.file?(file)
end

copy_export_files_to_destination_path
```

Reply:
@jbat how does this new approach seem? https://github.com/rdytech/superset-client/pull/40/commits/9b89c0b4e69f794cbf6b483efeee1b31ee9cf755